meeko/flexibility.py

```diff
@@ -5,275 +5,279 @@
+            and group_index != flex_model["root"]
+        ):
+            remove[group_index] = members[0]
+    for group_index, atom_index in remove.items():
+        host_index = graph[group_index][0]
+        flex_model["rigid_body_members"].pop(group_index)
+        flex_model["rigid_body_members"][host_index].append(atom_index)
+        flex_model["rigid_body_graph"].pop(group_index)
+        flex_model["rigid_body_graph"][host_index].remove(group_index)
+        flex_model["rigid_body_connectivity"].pop((host_index, group_index))
+        flex_model["rigid_body_connectivity"].pop((group_index, host_index))
+        flex_model["rigid_index_by_atom"][atom_index] = host_index
+    flex_model["rigid_body_count"] -= len(remove)
+    return
+
+
+def get_flexibility_model(
+    molsetup,
+    root_atom_index=None,
+    break_combo_data=None,
+):
+    # No macrocyclic rings are to be broken. We simply build the flexibility model.
+    if break_combo_data is None or len(break_combo_data["bond_break_combos"]) == 0:
+        bonds_to_break = ()
+        unbroken_rings_bonds = []
+        for ring in molsetup.rings:
+            for bond in molsetup.get_bonds_in_ring(ring):
+                unbroken_rings_bonds.append(bond)
+        flex_model = walk_rigid_body_graph(molsetup, bonds_to_break, unbroken_rings_bonds)
+        nr_not_ignored = sum(
+            [not molsetup.atom_ignore[i] for i in range(len(molsetup.atom_ignore))]
+        )
+        if len(flex_model["visited"]) != nr_not_ignored:
+            molsetup.show()
+            for key, value in flex_model.items():
+                print(key, value)
+            msg = f"{len(flex_model['visited'])=} differs from not-ignored atoms {nr_not_ignored}"
+            raise RuntimeError(msg)
+        root_body_index = get_root_body_index(flex_model, root_atom_index)
+        flex_model["root"] = root_body_index
+        broken_bonds = []
+        return flex_model, broken_bonds
+
+    # The macrocycle typer enumerated rings to break, and a number of lists
+    # of bonds to break/delete that result in breaking of the enumerated rings.
+    # to be deleted to break each ring. Deleting a bond results in the remaining
+    # bonds within the same ring to become rotatable.
+    bond_break_combos = break_combo_data["bond_break_combos"]
+    bond_break_scores = break_combo_data["bond_break_scores"]
+    unbroken_rings_list = break_combo_data["unbroken_rings"]
+    best_model = None
+    best_score = float("+inf")
+    best_index = None
+    for index in range(len(bond_break_combos)):
+        bond_break_combo = bond_break_combos[index]
+        bond_break_score = bond_break_scores[index]
+        unbroken_rings_bonds = []
+        for ring in unbroken_rings_list[index]:
+            for bond in molsetup.get_bonds_in_ring(ring):
+                unbroken_rings_bonds.append(bond)
+
+        model = walk_rigid_body_graph(molsetup, bond_break_combo, unbroken_rings_bonds)
+        nr_not_ignored = sum(
```

Comment:
I don't think this would be changing with the loop, but maybe I'm missing something. Could it just be instantiated outside the loop?